liionpack/netlist_utils.py

```diff
@@ -230,8 +230,12 @@ def setup_circuit(
                 desc.append("V" + str(num_V))
                 num_V += 1
                 val = V
-            node1.append(nodes[row + 1])
-            node2.append(nodes[row])
+            if rtype[row][0] == "R":
+                node1.append(nodes[row + 1])
+                node2.append(nodes[row])
+            else:
+                node1.append(nodes[row + 1])
+                node2.append(nodes[row])
```

Comment:
I don't get this. Both branches do the same thing?

Reply:
Jeez that was quick I think this was the net effect of me changing it and changing it back a different way so these lines can be removed. The import change in this PR is on the terminal node. I haven't quite finished making all the changes and I want to add an integration test

Reply:
Haha ok I'll wait until you're done

Reply:
You can take a look now

Reply:
There's a small difference between liionpack and pybamm around the discontinuities in current for the cycle steps. Not sure if this would be an issue for drive cycles so something to look out for. The calculation of the internal resistance is very simple and could maybe be improved around these events